### PR TITLE
slint-sc: Add spec ID annotations and tests for Slint SC language specification

### PR DESCRIPTION
**Pull request by ogoffart**
Annotate existing slint-sc tests with `//#sls.` comments linking them to the language specification paragraphs they exercise. Add new positive tests for source file structure, comments, whitespace, BOM encoding, element nesting, and identifiers.

### CODE REVIEW

File: internal/compiler/tests/syntax/slint-sc/empty_file.slint [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+// An empty source file (only comments) is well-formed.
+//#sls.file.source-file.empty
```
Comment (ogoffart) [inline internal/compiler/tests/syntax/slint-sc/empty_file.slint:5]: well, technically, the file is not empty :-/
hard to make a file without the copyright header and the //#sls comment, but maybe we can have another really empty file and reference it from here.

File: internal/compiler/tests/syntax/slint-sc/bom-encoding.slint [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+// A source file with a UTF-8 BOM is well-formed.
+//#sls.source.encoding.bom
```
Comment (ogoffart) [inline internal/compiler/tests/syntax/slint-sc/bom-encoding.slint:5]: note that just like for internal/compiler/tests/syntax/basic/bom-simple.slint, there is a test that check that it has the BOM https://github.com/slint-ui/slint/blob/e8dd6d11a9d08d2d835638c1bfae4f01042703cf/internal/compiler/tests/syntax_tests.rs#L100-L104

Comment (tronical) [reply internal/compiler/tests/syntax/slint-sc/bom-encoding.slint:5]: Cool idea to check the test :)

File: internal/compiler/tests/syntax/slint-sc/empty_file.slint (hunk 1 above)
Comment (tronical) [reply internal/compiler/tests/syntax/slint-sc/empty_file.slint:5]: Hah. Maybe a programmatic test would suffice here instead of a file in the test suite?

**Review by tronical — APPROVED** [review]
